pkg/controller/core.oam.dev/v1alpha2/application/assemble/assemble.go

```diff
@@ -322,7 +280,7 @@ func (am *AppManifests) assembleWorkload(comp *v1alpha2.Component, labels map[st
 	workloadType := wl.GetLabels()[oam.WorkloadTypeLabel]
 	compDefinition := am.AppRevision.Spec.ComponentDefinitions[workloadType]
 	for _, wo := range am.WorkloadOptions {
-		if err := wo.ApplyToWorkload(wl, comp.DeepCopy(), compDefinition.DeepCopy()); err != nil {
+		if err := wo.ApplyToWorkload(wl, compDefinition.DeepCopy(), addons); err != nil {
```

Comment:
please make sure if we need to create deepcopy for addons here.

Reply:
sure, will use deepcopy for addons.